consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/CMCConsumerGateway.java

```diff
@@ -164,15 +168,21 @@ public SubmitQueryResponse submitQuery(ISubmitQueryRequest submitQueryRequest,
         String xmlContent = SdtContext.getContext().getRawOutXml();
         String fromDate = xmlElementValueReader.getElementValue(xmlContent, FROM_DATE);
         String toDate = xmlElementValueReader.getElementValue(xmlContent, TO_DATE);
-        ClaimDefencesResponse response = claimDefences.claimDefences(idamRepository.getSdtSystemUserAccessToken(),
-                                                                     s2SRepository.getS2SToken(),
-                                                                     SdtContext.getContext().getCustomerIdamId(),
-                                                                     fromDate,
-                                                                     toDate);
-
         SubmitQueryResponse submitQueryResponse = new SubmitQueryResponse();
-        submitQueryResponse.setClaimDefencesResults(response.getResults());
-        submitQueryResponse.setClaimDefencesResultsCount(response.getResultCount());
+        try {
+            ClaimDefencesResponse response = claimDefences.claimDefences(idamRepository.getSdtSystemUserAccessToken(),
+                                                                         s2SRepository.getS2SToken(),
+                                                                         SdtContext.getContext().getCustomerIdamId(),
+                                                                         fromDate,
+                                                                         toDate);
+            submitQueryResponse.setClaimDefencesResults(response.getResults());
+            submitQueryResponse.setClaimDefencesResultsCount(response.getResultCount());
+            submitQueryRequest.setErrorLog(null);
+            submitQueryRequest.setStatus(StatusCodeType.OK.value());
```

Comment:
This will also overwrite status 78 (Results truncated - maximum number allowed reached) that could be returned by the first ConsumerGateway.  This shouldn't be overwritten as it means the users won't be aware that there were more results that matched their query.

It may be better to not call CMCConsumerGateway if ConsumerGateway returns status 78.  Calling CMCConsumerGateway in this situation would never reduce the number of results, so status 78 would still be true.

There's also the situation where the individual results from both ConsumerGateways don't breach the limit, but when combined they do.  Should there be an overall limit, or one for each gateway?

Reply:
CMCConsumerGateway no longer called if ConsumerGateway returns status code 78.  There are some other issues though which have documented in the class comment.